src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -5447,6 +5447,49 @@ class spell_gen_face_rage : public AuraScript
     }
 };
 
+enum Shadowmeld {
+
+    SPELL_RACIAL_ELUSIVENESS = 21009
+};
+
+// 58984 Shadowmeld (Racial)
+class spell_gen_shadowmeld : public AuraScript {
+
+    bool Validate(SpellInfo const* /*spell*/) override
+    {
+        return ValidateSpellInfo({ SPELL_RACIAL_ELUSIVENESS });
+    }
+
+    void OnApply(AuraEffect const* /*effect*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* caster = GetCaster();
+
+        if (AuraEffect const* aurEff = caster->GetAuraEffect(SPELL_RACIAL_ELUSIVENESS, EFFECT_0)) {
+
+            int32 basepoints0 = aurEff->GetAmount();
+            StealthType type = StealthType(aurEff->GetMiscValue());
+            caster->m_stealth.AddValue(type, basepoints0);
+        }
+    }
+
+    void OnRemove(AuraEffect const* /*effect*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* caster = GetCaster();
+
```

Comment:
this is a hack. Shadowmeld uses a stealth aura effect which you can manipulate by using the OnEffectCalcValue hook. There you can add the value of dummy effect to the already calculated value

Reply:
Hmm, so I have to do something like
`spellInfo->Effects[EFFECT_2].CalcValue(GetCaster(), basepoints0);` ?

Reply:
https://github.com/The-Cataclysm-Preservation-Project/TrinityCore/commit/6e412b16de04d7cd9ef7efa8b75b5ffd87db70f7 something like this.